validator/src/validator.rs

```diff
@@ -299,6 +300,20 @@ where
         spawn(async move {
             network.set_validators(voting_keys).await;
         });
+
+        // Check validator configuration
+        let _ = self.get_validator(&blockchain)
```

Comment:
```suggestion
        if let Ok(Some(validator)) = self.get_validator(&blockchain) {
            ...
        }
```

As you are only interested in the `if let Ok(Some(validator))` case, this is basically the same without the nested `map`s.